drivers/ksfilter/ks/irp.c

```diff
@@ -2032,6 +2023,9 @@ KsSetMajorFunctionHandler(
 {
     DPRINT("KsSetMajorFunctionHandler Function %x\n", MajorFunction);
 
+    // HACK for MS portcls
+    DriverObject->MajorFunction[IRP_MJ_CREATE] = KspCreate;
+
```

Comment:
Can it be because we don't handle KSDISPATCH_FASTIO flag ? According to MSDN, this flag "may be added to the MajorFunction identifier".
It should be a matter of setting function pointers in DriverEntry->FastIoDispatch.

Reply:
> Can it be because we don't handle KSDISPATCH_FASTIO flag ?

I can't say exactly, but most likely no, because MSDN also says that "This is valid only with IRP_MJ_READ, IRP_MJ_WRITE or IRP_MJ_DEVICE_CONTROL." In our case, driver calls `IRP_MJ_CREATE`, so that flag seems not be used.